File: backend/app/schemas/appointment.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import AppointmentStatus


class CreateAppointmentRequest(BaseModel):
    patient_id: uuid.UUID
    chief_complaint: str | None = Field(default=None, max_length=512)
    scheduled_at: datetime | None = None


class AppointmentSummary(BaseModel):
    id: uuid.UUID
    patient_id: uuid.UUID
    nurse_id: uuid.UUID
    doctor_id: uuid.UUID | None
    status: AppointmentStatus
    chief_complaint: str | None
    scheduled_at: datetime | None
    intake_completed_at: datetime | None
    prescription_completed_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/appointment.ts
// Mirrors schemas/appointment.py AppointmentSummary + enums.py AppointmentStatus.

export type AppointmentStatus =
  | "created"
  | "intake_in_progress"
  | "intake_complete"
  | "with_doctor"
  | "prescription_complete"
  | "complete";

export interface Appointment {
  id: string;
  patient_id: string;
  nurse_id: string;
  doctor_id: string | null;
  status: AppointmentStatus;
  chief_complaint: string | null;
  scheduled_at: string | null;
  intake_completed_at: string | null;
  prescription_completed_at: string | null;
  created_at: string;
}

export interface CreateAppointmentPayload {
  patient_id: string;
  chief_complaint?: string;
  scheduled_at?: string;
}